### PR TITLE
fix(ts): better `tsconfig.json` handling & improve tests

### PR DESCRIPTION
**Pull request by kevinmarrec**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
I refactored code and fixed `ts-node` registration by telling him where is the right `tsconfig.json`.

## Checklist:
- [x] I have updated tests to cover my changes (if not applicable, please state why)
- [x] All updated and existing tests are passing.

### CODE REVIEW

File: distributions/nuxt-ts/bin/nuxt-ts.js [hunk 1]
```diff
@@ -5,10 +5,11 @@ process.env.NUXT_TS = 'true'
 
 const nuxtCommands = ['dev', 'build', 'generate', 'start']
 const rootDir = (process.argv[2] && !nuxtCommands.includes(process.argv[2])) ? process.argv[2] : process.cwd()
+const tsConfigPath = require('path').resolve(rootDir, 'tsconfig.json')
```
Comment (pi0) [inline distributions/nuxt-ts/bin/nuxt-ts.js]: Prefer moving require/import to top of the script, when lazy loading is not happening

File: packages/typescript/src/index.js [hunk 2]
```diff
@@ -1,7 +1,42 @@
...
+        '@nuxt/vue-app'
+      ]
+    }
+  }, { spaces: 2 })
+  consola.info(`Extending ${chalk.bold.blue(`node_modules/${configToExtend}/tsconfig.json`)}`)
+  consola.success(`Generated successfully at ${chalk.bold.green(tsConfigPath)}`)
+}
+
+export async function setup(tsConfigPath) {
+  if (!existsSync(tsConfigPath)) {
```
Comment (pi0) [inline packages/typescript/src/index.js]: Can we add a guard to prevent calling function multiple times?

**Review by pi0 — DISMISSED** [review]

File: distributions/nuxt-ts/bin/nuxt-ts.js (hunk 1 above)
Comment (kevinmarrec) [reply distributions/nuxt-ts/bin/nuxt-ts.js]: I don't think `import { resolve } from 'path'`  will work as it's the binary and not compiled.

So just `const { resolve } = require('path')` at the top should be ok ?

Comment (pi0) [reply distributions/nuxt-ts/bin/nuxt-ts.js]: @kevinmarrec YES :)

File: packages/typescript/src/index.js (hunk 2 above)
Comment (kevinmarrec) [reply packages/typescript/src/index.js]: What could be the case when it's called multiple times ?

How do we implement such `guard` ? :)

Comment (pi0) [reply packages/typescript/src/index.js]: Like this:

https://github.com/nuxt/nuxt.js/blob/dev/packages/cli/src/setup.js#L13

> What could be the case when it's called multiple times?

TS Node will be registered twice :)

Comment (kevinmarrec) [reply packages/typescript/src/index.js]: I was meaning "In which case `setup` will be called twice ?", but I'll put a guard anyway ;)

Comment (kevinmarrec) [reply packages/typescript/src/index.js]: Thanks for the code ref

Comment (pi0) [reply packages/typescript/src/index.js]: Programmatic usage or when forking workers. But generally my reason is ensurance :)

Comment (kevinmarrec) [reply packages/typescript/src/index.js]: If I define a `_setup` variable, can it conflicts with `_setup` of CLI ? Or is `let` make it defined at the file scope ?

**Comment by kevinmarrec** [comment]
@pi0 Gonna edit tests, I think I'm doing something bad as i'm not awaiting `setup` in `beforeAll`, and will also add a short test to test the guard :)

File: packages/typescript/src/index.js (hunk 2 above)
Comment (pi0) [reply packages/typescript/src/index.js]: @kevinmarrec rollup takes care of renaming even if they are bundled. No worries about same variable name or scope.

Comment (kevinmarrec) [reply packages/typescript/src/index.js]: Perfect then, thanks for the info 💪

**Comment by clarkdo** [comment]
@kevinmarrec The pr looks great.

Could you move all typescript package related tests into `packages/typescript/test` dir.

**Comment by kevinmarrec** [comment]
Will do, thanks @clarkdo

**Comment by kevinmarrec** [comment]
@clarkdo Should we also move `typescript` fixtures under `packages/typescript/test/fixtures` ?
Could be cleaner to put everything TS related things around this package now we have it.

**Comment by kevinmarrec** [comment]
I would also want to remove `typescript-custom`, I consider it obsolete as we've built-in TS support.

**Comment by clarkdo** [comment]
I think we only need unit tests, fixtures will be removed in future

**Comment by clarkdo** [comment]
So for now it would be great if we write unit tests which can cover 100% packages/typescript

**Comment by kevinmarrec** [comment]
@clarkdo 100% coverage right now, but not 100% branches covered, I probably need to mock `existsSync` and enquirer `prompt` to get 100% branches covered

**Review by pi0 — APPROVED** [review]

**Pull request merged**